test/common/testlib.py

```diff
@@ -1169,6 +1174,19 @@ def skipPackage(*args):
     return lambda func: func
 
 
+def nonDestructive(*args):
```

Comment:
why not just `test_entity` directly as the lone argument to the function?

if you want to allow more possibilities later, then you can do that later...